### DISCUSSION TITLE
Set fft_size per Device?

### DISCUSSION THREAD

**Discussion by thowe-switzerland** [rtl-airband/RTLSDR-Airband#290] [General]
Hello all

If I see it correctly, the fft_size is currently a global option which applies to all devices. But now I have the challenge that I would like to combine several devices with different bandwidths: 3MHz and 6MHz. But I would also like to double the fft_size for the 6MHz so that the channel width is about the same.

Is there a reason that the fft_size is a global option and cannot be set per device?

Best thanks & greetings
Thomas

**Comment by charlie-foxtrot** [comment]
`fit_size` is used as a global in the code so it would take a good amount of refactoring to make it per-device.  With that said, I don't there is anything preventing it if you want to try.  You can always run two instances of `rtl_airband` in the meantime.

**Reply by thowe-switzerland** [reply]
Thank you. If nothing speaks against it I will look at the appropriate opportunity whether I can adapt that.

Two separate instances don't work here, because channels from both devices should flow into one mixer.
